src/main/java/gg/solarmc/clans/events/ChatEvent.java

```diff
@@ -0,0 +1,51 @@
+package gg.solarmc.clans.events;
+
+import gg.solarmc.clans.ChatMode;
+import gg.solarmc.clans.helper.ChatHelper;
+import gg.solarmc.clans.helper.PluginHelper;
+import gg.solarmc.loader.clans.Clan;
+import gg.solarmc.loader.clans.ClansKey;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.format.NamedTextColor;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.AsyncPlayerChatEvent;
+
+public class ChatEvent implements Listener {
+    private final ChatHelper chatHelper;
+    private final PluginHelper pluginHelper;
+
+    public ChatEvent(PluginHelper pluginHelper, ChatHelper chatHelper) {
+        this.chatHelper = chatHelper;
+        this.pluginHelper = pluginHelper;
+    }
+
+    @EventHandler
```

Comment:
Are you sure this handler runs before deluxechat's? If so, you should comment that in, because it's important to the maintenance of the code

Reply:
No i am not sure , i'll check when i am testing it